test/prob_kwargs.jl

```diff
@@ -29,3 +29,37 @@ a = ones(3)
 @test Zygote.gradient(f, a)[1][1] ≈ Zygote.gradient(f2, a)[1][1]
 @test Zygote.gradient(f, a)[1][2] == Zygote.gradient(f2, a)[1][2] == 0
 @test Zygote.gradient(f, a)[1][3] == Zygote.gradient(f2, a)[1][3] == 0
+
+# callback in problem construction or in solve call should give same result
+# https://github.com/SciML/SciMLSensitivity.jl/issues/1081
+odef(du, u, p, t) = du .= u .* p
+prob = ODEProblem(odef, [2.0], (0.0, 1.0), [3.0])
+
+let callback_count1 = 0, callback_count2 = 0
+    function f1(u0p, adjoint_type)
+        condition(u, t, integrator) = t == 0.5
+        affect!(integrator) = callback_count1 += 1
+        cb = DiscreteCallback(condition, affect!)
+        prob = ODEProblem{true}(odef, u0p[1:1], (0.0, 1.0), u0p[2:2]; callback = cb)
+        sum(solve(prob, Tsit5(), tstops = [0.5], sensealg = adjoint_type))
+    end
+
+    function f2(u0p, adjoint_type)
+        condition(u, t, integrator) = t == 0.5
+        affect!(integrator) = callback_count2 += 1
+        cb = DiscreteCallback(condition, affect!)
+        prob = ODEProblem{true}(odef, u0p[1:1], (0.0, 1.0), u0p[2:2])
+        sum(solve(prob, Tsit5(), tstops = [0.5], callback = cb, sensealg = adjoint_type))
+    end
+
+    @testset "Callback duplication check" begin
+        for adjoint_type in [
+            ForwardDiffSensitivity(), ReverseDiffAdjoint(), TrackerAdjoint()]
```

Comment:
and loop through some adjoint methods too, all sensealgs is best

Reply:
Except ZygoteAdjoint

Reply:
`ZygoteAdjoint()` and `GaussAdjoint()` error out, besides those I've included all the rest that support callbacks

Reply:
For the ones that error, label them as @test_broken. 

Gauss adjoint with callbacks is currently on the critical Todo.